art/attacks/evasion/projected_gradient_descent/projected_gradient_descent.py

```diff
@@ -191,11 +191,37 @@ def _check_params(self) -> None:
         if self.norm not in [1, 2, np.inf, "inf"]:
             raise ValueError('Norm order must be either 1, 2, `np.inf` or "inf".')
 
-        if self.eps <= 0:
-            raise ValueError("The perturbation size `eps` has to be positive.")
+        if (not (isinstance(self.eps, float) and isinstance(self.eps_step, float))) and (
+            not (isinstance(self.eps, np.ndarray) and isinstance(self.eps_step, np.ndarray))
+        ):
+            raise TypeError(
+                "The perturbation size `eps` and the perturbation step-size `eps_step` must have the same type."
+            )
+
+        if isinstance(self.eps, float):
+            if self.eps <= 0:
+                raise ValueError("The perturbation size `eps` has to be positive.")
+
+            if self.eps_step <= 0:
+                raise ValueError("The perturbation step-size `eps_step` has to be positive.")
 
-        if self.eps_step <= 0:
-            raise ValueError("The perturbation step-size `eps_step` has to be positive.")
+            if self.eps_step > self.eps:
+                raise ValueError("The iteration step `eps_step` has to be smaller than the total attack `eps`.")
+
+        else:
+            if (self.eps <= 0).any():
+                raise ValueError("The perturbation size `eps` has to be positive.")
+
+            if (self.eps_step <= 0).any():
+                raise ValueError("The perturbation step-size `eps_step` has to be positive.")
+
+            if self.eps.shape != self.eps_step.shape:
+                raise ValueError(
+                    "The perturbation size `eps` and the perturbation step-size `eps_step` must have the same shape."
+                )
+
+            if (self.eps_step > self.eps).any():
+                raise ValueError("The iteration step `eps_step` has to be smaller than the total attack `eps`.")
```

Comment:
I think this might only be true if `norm = np.inf` as in the removed check in line 212.

Reply:
solved.